src/lib.rs

```diff
@@ -161,8 +163,9 @@ impl DecodeReaderBytesBuilder {
         }
         let encoding = self.encoding
             .map(|enc| enc.new_decoder_with_bom_removal());
-        // No need to do BOM detection if we have an explicit encoding.
-        let has_detected = !self.bom_override && encoding.is_some();
+
+        // No need to do BOM detection if we opt out of it or have an explicit encoding.
+        let has_detected = !self.bom_sniffing || (!self.bom_override && encoding.is_some());
```

Comment:
Sorry, I just noticed this: can you make sure all new code introduced in this is wrapped to 79 columns inclusive? It should follow the existing style of the code. This line in particular is longer than 79.

Reply:
Done. I wanted to use `cargo fmt` at first but didn't know if this repository follows it's rules. Perhaps this tool can be used in the future?

Reply:
It'd be nice, but no, I don't think rustfmt is in a place yet for me to adopt it. It does too many sub-optimal things with the formatting.